### DISCUSSION TITLE
Changed event sequence for HTTP 1.0 responses without content-length

### DISCUSSION THREAD

**Discussion by fadam-asis** [cesanta/mongoose#1475] [Ask questions, get answers] [answered]
We recently updated from version 7.4 to 7.6 and discovered an unexpected
change in the sequence of event calls to the user callback function.

When sending an HTTP request that is answered with an HTTP 1.0 response
without a content-length header field, the event MG_EV_CLOSE is triggered
before the MG_EV_HTTP_MSG.

Event sequence before the change:
  1. MG_EV_HTTP_CHUNK
  2. MG_EV_HTTP_MSG
  3. MG_EV_CLOSE

Sequence after the change:
  1. MG_EV_HTTP_CHUNK
  2. MG_EV_CLOSE
  3. MG_EV_HTTP_MSG

We discovered that this is the result of the switched function call
in mg_call that was done with commit 03d187b1 (https://github.com/cesanta/mongoose/pull/1388)
and wondered if this is a intended behavior?

**Comment by cpq** [comment, marked as answer]
@fadam-asis that is an unwanted side-effect from the aforementioned change.
The message sequence should be as it was, at the same time the callback invocation sequence should be as it is.
If you have ideas how to achieve that, please share.
